packages/vitest/src/runtime/mocker.ts

```diff
@@ -336,6 +354,16 @@ export class VitestMocker {
   }
 
   public queueMock(id: string, importer: string, factory?: () => unknown) {
+    const factoryHandler = {
+      apply(target: () => unknown) {
+        const value = target()
+        if (typeof value !== 'object')
+          throw new Error(`[vitest] vi.mock("${id}", factory?: () => unknown) is not returning an object. Did you mean to return an object with a "default" key?`)
+        return value
+      },
+    }
```

Comment:
Why do we need a proxy, if we can just check it here:
https://github.com/jereklas/vitest/blob/bdd74e50836a70d9aae410781f5c6fdbe5093f3a/packages/vitest/src/runtime/mocker.ts#L109
?

Reply:
You need the proxy on the exports object because you don't know which export is being accessed until the calling code imports something?

I need to head to bed, but will try taking a look at your factory related concern tomorrow.

Reply:
We do need a roxy for exports object, but I don't see why we need proxy for calling a factory - we are calling it ourselves on https://github.com/jereklas/vitest/blob/bdd74e50836a70d9aae410781f5c6fdbe5093f3a/packages/vitest/src/runtime/mocker.ts#L109

Reply:
Ah gotcha. I misunderstood your statement.